### PR TITLE
add argument --allow-folder-id-mismatch to report addon_xml_matches_folder as INFO

### PR DESCRIPTION
**Pull request by anxdpanic**
w/ arg:
```INFO: Addon id and folder name does not match. Ensure folder name is <addon_id> when submitting a PR to Kodi's official repository.```

w/o arg:
```ERROR: Addon id and folder name does not match.```

### CODE REVIEW

File: kodi_addon_checker/__main__.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def check_artifact(artifact_path, args, branch_name, all_repo_addons, pr):
     config = Config(artifact_path, args)
     ConfigManager.process_config(config)
     if os.path.isfile(os.path.join(artifact_path, "addon.xml")):
-        return check_addon.start(artifact_path, branch_name, all_repo_addons, pr, config)
+        return check_addon.start(artifact_path, branch_name, all_repo_addons, pr, not pr, config)
```
Comment (Razzeee) [inline kodi_addon_checker/__main__.py]: Not a fan of `sa` as the combination of `pr` and `sa` are not all handled (or available).
Any reason your not using `pr` directly?

**Comment by Razzeee** [comment]
Should be fine for the kodi repo usecases, but probably won't work for the ci of single repos.

**Comment by anxdpanic** [comment]
Should I change this to have it's own argument ```--skip-folder-matches-id``` and change the warning to info?

**Comment by Rechi** [comment]
Yes it should be changed to have it's own argument.
The name `--skip-folder-matches-id` would make me think that actually executing the `addon_xml_matches_folder` function is completely skipped, not that reports a warning or info instead of an error. But that's nothing I care much about.

**Comment by anxdpanic** [comment]
Any feelings on warning vs info? I went with warning here since it's a must when submitting.

**Comment by Rechi** [comment]
The following line needs an update too.
https://github.com/xbmc/addon-check/blob/ad8aa4960a0fefdb2f4661ff9fd7f2e90427e445/tests/test_check_addon.py#L26

**Comment by anxdpanic** [comment]
Updated test_check_addon, required adding an ```Args``` class as well

File: kodi_addon_checker/__main__.py [hunk 2]
```diff
@@ -74,6 +74,7 @@ def main():
     parser.add_argument("--branch", choices=choice, required=True,
                         help="Target branch name where the checker will resolve dependencies")
     parser.add_argument("--PR", help="Tell if tool is to run on a pull requests or not", action='store_true')
+    parser.add_argument("--allow-folder-id-mismatch", help="Allow the addon's folder name and id to mismatch", action="store_true")
```
Comment (Rechi) [inline kodi_addon_checker/__main__.py]: > ./kodi_addon_checker/__main__.py:77:121: E501 line too long (131 > 120 characters)

**Comment by Razzeee** [comment]
The new argument needs to be documented in the readme

**Comment by anxdpanic** [comment]
Added to the readme and shifted the description alignment accordingly

**Comment by Razzeee** [comment]
thank you!

**Pull request merged**